src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -84,6 +89,21 @@ public MetricsDB(long windowStartTime) throws Exception {
     create = DSL.using(conn, SQLDialect.SQLITE);
   }
 
+  /**
+   * Returns a MetricsDB handle associated with an existing metricsdb file.
+   *
+   * @param windowStartTime the timestamp associated with an existing metricsdb file
+   * @return a MetricsDB handle associated with the metricsdb file
+   * @throws Exception if the metricsdb file does not exist or is invalid
+   */
+  public static MetricsDB fetchExisting(long windowStartTime) throws Exception {
```

Comment:
It would not. This method makes use of the MetricsDB ctor, which throws an Exception if there is some issue creating a connection to that file (like if the file is corrupt or something).